packages/features/ee/teams/components/TeamsUpgradeBanner.tsx

```diff
@@ -0,0 +1,38 @@
+import { useRouter } from "next/router";
+
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { trpc } from "@calcom/trpc/react";
+import { TopBanner } from "@calcom/ui/components";
+import { showToast } from "@calcom/ui/v2/core";
+
+export function TeamsUpgradeBanner() {
+  const { t } = useLocale();
+  const router = useRouter();
+  const { data } = trpc.viewer.teams.getUpgradeable.useQuery();
+  const publishTeamMutation = trpc.viewer.teams.publish.useMutation({
+    onSuccess(data) {
+      router.push(data.url);
+    },
+    onError: (error) => {
+      showToast(error.message, "error");
+    },
+  });
+  if (!data) return null;
+  const [team] = data;
+  if (!team) return null;
```

Comment:
Instead of displaying all unpublished teams at once. We get the first one and ask for upgrade. Rinse and repeat until all teams are upgraded or deleted.